src/Patches/PlayerCustomizationDummyPatch.cs

```diff
@@ -0,0 +1,45 @@
+﻿using HarmonyLib;
+using System.Collections.Generic;
+using UnityEngine;
+
+using Plugin = MoreCustomizations.MoreCustomizationsPlugin;
+
+namespace MoreCustomizations.Patches;
+
+public class PlayerCustomizationDummyPatch {
+    [HarmonyPatch(typeof(PlayerCustomizationDummy), "SetPlayerHat")]
+    [HarmonyPrefix]
+    private static void SetPlayerHat(PlayerCustomizationDummy __instance, int index) {
+        if (Character.localCharacter == null)
+            return;
+
+        CharacterCustomization characterCustomization = Character.localCharacter.gameObject.GetComponent<CharacterCustomization>();
+        if (characterCustomization == null)
+            return;
+
+        // check if character hats more than dummy hats
+        if (characterCustomization.refs.playerHats.Length > __instance.refs.playerHats.Length) {
+            Plugin.Logger.LogInfo($"{characterCustomization.refs.playerHats.Length}-{__instance.refs.playerHats.Length}");
```

Comment:
If this is debugging purposes, I suggest to delete this line, or add some context for logging like below.
```csharp
Plugin.Logger.LogInfo($"Dummy hat count difference : {characterCustomization.refs.playerHats.Length} - {__instance.refs.playerHats.Length}");